framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -3236,6 +3376,18 @@ HRESULT Dx12ReplayConsumerBase::OverrideCreateReservedResource(
     if (SUCCEEDED(replay_result))
     {
         SetIsReservedResource(resource);
+
+        auto res        = static_cast<ID3D12Resource*>(*resource->GetHandlePointer());
+        auto extra_info = std::make_unique<D3D12ResourceInfo>();
+
+        extra_info->subresource_count = graphics::Dx12ResourceDataUtil::GetSubresourceCount(res);
+        extra_info->resource_state_infos.resize(extra_info->subresource_count);
+        for (auto& info : extra_info->resource_state_infos)
+        {
+            info.states        = initial_state;
+            info.barrier_flags = D3D12_RESOURCE_BARRIER_FLAG_NONE;
+        }
+        SetExtraInfo(resource, std::move(extra_info));
```

Comment:
Also, `SetIsReservedResource(resource);` will create an `D3D12ResourceInfo`, and then it looks like this code will also create another one and replace the one created by `SetIsReservedResource`.

Reply:
Another idea: `Dx12ReplayConsumerBase::SetResourceDesc()` is called from every resource creation function and is added automatically to the generated code in `generated_dx12_replay_consumer.cpp`. Maybe this functionality could be moved there. This might also avoid the need to create override functions for the other resource types.

Reply:
I added `InitialResourceExtraInfo`. `SetIsReservedResource` also was moved to there.  Some data can't be gotten in `SetResourceDesc`.